sdk/log/record.go

```diff
@@ -131,6 +150,24 @@ func (r *Record) WalkAttributes(f func(log.KeyValue) bool) {
 
 // AddAttributes adds attributes to the log record.
 func (r *Record) AddAttributes(attrs ...log.KeyValue) {
+	if r.attributeCountLimit > 0 && r.AttributesLen()+len(attrs) > r.attributeCountLimit {
+		r.compactAttr()
```

Comment:
shouldn't attributes already be compacted here?

Reply:
Yeah, the current pushed version that is not a grantee given this conditional is not always executed. The changes I'm looking at pushing to address your other comment will ensure they are.

Stay tuned ... :smile:

Reply:
Updated.